tests/models/test_message.py

```diff
@@ -0,0 +1,209 @@
+    assert file.is_opn() is expected
+
+
+def test_file_from_event(dd_event):
+    file = File.from_event(dd_event("OPN2102R"))
+    assert file.name == "dd_OPN2102R_0103202021_16428.zip:ons-blaise-v2-nifi"
+    assert file.sizeBytes == "20"
+    assert file.md5sum == "d1ad7875be9ee3c6fde3b6f9efdf3c6b67fad78ebd7f6dbc"
+    assert file.relativePath == ".\\"
+
+
+def test_create_message_for_management_information(mi_event, config):
+    actual_message = create_message(mi_event, config)
+    assert (
+        actual_message.description
+        == "Management Information files uploaded to GCP bucket from Blaise5"
+    )
+    assert actual_message.dataset == "blaise_mi"
+    assert actual_message.iterationL1 == "survey_on_prem_subfolder"
+    assert actual_message.iterationL2 == ""
+
+
+def test_create_message_for_data_delivery_opn(dd_event, config):
+    dd_event = dd_event("OPN2101A")
+    actual_message = create_message(dd_event, config)
+
+    assert (
+        actual_message.description
+        == "Data Delivery files for OPN uploaded to GCP bucket from Blaise5"
+    )
+    assert actual_message.dataset == "blaise_dde"
+    assert actual_message.iterationL1 == "SYSTEMS"
+    assert actual_message.iterationL2 == "survey_on_prem_subfolder"
+    assert actual_message.iterationL3 == "OPN"
+    assert actual_message.iterationL4 == "OPN2101A"
+
+
+@pytest.mark.parametrize(
+    "instrument,expected_survey_name",
+    [
+        ("LMS2102_A1", "LMS"),
+        ("lms2102_bk1", "LMS"),
+        ("lmc2102_bk1", "LMC"),
+        ("lmb21021_bk2", "LMB"),
+    ],
+)
+def test_create_message_for_data_delivery_lms(
+    instrument, expected_survey_name, dd_event, config
+):
+    dd_event = dd_event(instrument)
+    actual_message = create_message(dd_event, config)
+
+    assert (
+        actual_message.description
+        == f"Data Delivery files for {expected_survey_name} uploaded to GCP bucket from Blaise5"
+    )
+    assert actual_message.dataset == "blaise_dde_lms"
+    assert actual_message.iterationL1 == "CLOUD"
+    assert actual_message.iterationL2 == "test"
+    assert actual_message.iterationL3 == instrument.upper()
+    assert actual_message.iterationL4 == ""
+
+
+@pytest.mark.parametrize(
+    "spicy_file_extension",
+    [
+        ("avi"),
+        ("dat"),
+        ("nth"),
+        ("zoo"),
+        ("qxd"),
+    ],
+)
+def test_create_message_with_an_invalid_file_extension(
+    spicy_file_extension, dd_event, config
+):
+    dd_event = dd_event("OPN2101A")
+    dd_event["name"] = f"dd_opn2101a.{spicy_file_extension}:my-bucket-name"
+
+    with pytest.raises(InvalidFileExtension):
+        create_message(dd_event, config)
+
+
+@pytest.mark.parametrize(
+    "spicy_file_types",
+    [
+        ("notMI"),
+        ("notDD"),
+        ("ddfoo"),
+        ("mibar"),
+        ("mmmm_spicy"),
+    ],
+)
+def test_create_message_with_an_invalid_file_type(spicy_file_types, dd_event, config):
+    dd_event = dd_event("OPN2101A")
```

Comment:
the event should probably be the `spicy_file_type` instead of `OPN2101A` (to avoid confusion)

Reply:
💯